src/plugins/allPlugins.ts

```diff
@@ -1,13 +1,24 @@
+import WebSocket from 'ws'
+
 import { serverConfig } from '../serverConfig'
 import { AddressPlugin } from '../types/addressPlugin'
-import { BlockbookOptions, makeBlockbook } from './blockbook'
+import { makeBlockbook } from './blockbook'
 import { makeFakePlugin } from './fakePlugin'
 
-function makeNowNode(opts: BlockbookOptions): AddressPlugin {
+export interface NowNodeOptions {
+  pluginId: string
+
+  /** The connection to blockbook */
+  url: string
+}
+
+function makeNowNode(opts: NowNodeOptions): AddressPlugin {
+  const url = `${opts.url}/${serverConfig.nowNodesApiKey}`
+  const ws = new WebSocket(url)
   return makeBlockbook({
     ...opts,
-    safeUrl: opts.url,
-    url: opts.url + '/' + serverConfig.nowNodesApiKey
+    url: opts.url,
+    ws
```

Comment:
The blockbook plugin needs to handle connection & disconnection, since Blockbook can come & go all day long. We don't want somebody at ChangeNow rebooting a server, and killing our change-server as well. We just need to reconnect.

See https://github.com/EdgeApp/edge-core-js/blob/42bd9172cf20927dde82d9848675b0ceba108d9d/src/core/currency/change-server-connection.ts for how to bundle the WebSocket & codec together so the plugin can just call `connectBlockbook` or whatever to get a new connection, if the old one dies.